src/memray/_memray/record_reader.cpp

```diff
@@ -194,7 +194,31 @@ RecordReader::parseAllocationRecord()
 
     auto& stack = d_stack_traces[record.tid];
     d_allocation_records.emplace_back(Allocation{
-            .record = record,
+            .tid = record.tid,
+            .address = record.address,
+            .size = record.size,
+            .allocator = record.allocator,
+            .native_frame_id = 0,
+            .frame_index = stack.empty() ? 0 : stack.back(),
+            .native_segment_generation = d_symbol_resolver.currentSegmentGeneration()});
+    return true;
+}
+
+bool
+RecordReader::parseNativeAllocationRecord()
+{
+    NativeAllocationRecord record;
+    if (!d_input->read(reinterpret_cast<char*>(&record), sizeof(NativeAllocationRecord))) {
```

Comment:
I'd prefer `sizeof(record)` since we're modifying this code - the `(&obj, sizeof(obj))` pattern is recognizable and eliminates the risk of updating the type of `obj` in one place and forgetting to update the `sizeof` to match.